pkgs/applications/version-management/gex/default.nix

```diff
@@ -1,21 +1,26 @@
 { lib
 , stdenv
+, openssl
+, pkg-config
 , rustPlatform
 , fetchFromGitHub
 }:
 
 rustPlatform.buildRustPackage rec {
   pname = "gex";
-  version = "0.3.3";
+  version = "0.3.8";
 
   src = fetchFromGitHub {
     owner = "Piturnah";
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-oUcQKpZqqb8wZDpdFfpxLpwdfQlokJE5bsoPwxh+JMM=";
+    hash = "sha256-pjyS0H25wdcexpzZ2vVzGTwDPzyvA9PDgzz81yLGTOY=";
   };
 
-  cargoHash = "sha256-ZFrIlNysjlXI8n78N2Hkff6gAplipxSQXUWG8HJq8fs=";
+  cargoHash = "sha256-+FwXm3QN9bt//dWqzkBzsGigyl1SSY4/P29QtV75V6M=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ];
```

Comment:
Shouldn't it be in `nativeBuildInputs` as well ?

Reply:
I'm very new to NixOS so I'm not sure, having trouble even understanding the purpose of `nativeBuildInputs` vs `buildInputs`.

I based this on the inputs for [gitoxide](https://github.com/NixOS/nixpkgs/blob/nixos-23.05/pkgs/applications/version-management/gitoxide/default.nix) which also takes a dependency on openssl, and it allowed the derivation to build fine on my system.

How can I test if it needs to be also in `nativeBuildInputs` for other systems?

Reply:
A quick search on Google : https://discourse.nixos.org/t/use-buildinputs-or-nativebuildinputs-for-nix-shell/8464

Try to move openssl in nativebuildinputs and test your derivation to see if it works.

Reply:
I also found this page, and I don't think `openssl` is used at build time, unless linking against it counts.

I tried moving `openssl` to `nativeBuildInputs` and it now fails to build the derivation.

It works both with `openssl` in `buildInputs` only and with it in both, but as per my first point, I'm not sure that it's necessary for it to be present in both.